embeddings/hyperparameter_search/lighting_configspace.py

```diff
@@ -0,0 +1,124 @@
+from dataclasses import InitVar, dataclass, field
+from typing import Dict, Final, List, Union
+
+from embeddings.hyperparameter_search.configspace import (
+    BaseConfigSpace,
+    Parameter,
+    SampledParameters,
+)
+from embeddings.hyperparameter_search.parameters import (
+    ConstantParameter,
+    ParameterValues,
+    SearchableParameter,
+)
+
+
+@dataclass
+class LightingTextClassificationConfigSpace(BaseConfigSpace):
+    embedding_name: InitVar[Union[str, List[str]]]
+    devices: InitVar[Union[int, str, None, List[int]]] = field(default="auto")
+    accelerator: InitVar[Union[str, None]] = field(default="auto")
+
+    param_embedding_name: Parameter = field(init=False)
+    trainer_devices: Parameter = field(init=False)
+    trainer_accelerator: Parameter = field(init=False)
+
+    max_epochs: Parameter = SearchableParameter(
+        name="max_epochs", type="int_uniform", low=1, high=30, step=1
+    )
+    mini_batch_size: Parameter = SearchableParameter(
+        name="batch_size", type="log_int_uniform", low=8, high=64, step=1
+    )
+    max_seq_length: Parameter = ConstantParameter(
```

Comment:
why is this constant parameter?

Reply:
To limit space for search, It can be overridden by the user if otherwise.

Reply:
Ok I see, but I'd still make this one a SearchableParameter, just with fewer values, and limit the pool of other parameters like finetune_last_n_layers. Setting only this one parameter as constant does not seem to be consistent.

Reply:
Why? We have limited resources to fine tune all parameters, so only part of it should be tuned. I would update configspaces after performing experiments to choose best performing ones based on the results.

Reply:
I would transfer all concerns with ConfigSpace setup to separate issue, after the experiments will be finished. WDYT? @djaniak @asawczyn From my point of view it can be any value set here as for now.

Reply:
Ok let's do that, here i don't like that we cherry picking this specific parameter to not tune, especially that we pick value for which the training takes the longest